### ISSUE TITLE
Error handling should be based on Errors.Is () from stdlib errors

### ISSUE THREAD

**Issue by zibi94** [volatiletech/sqlboiler#1080]
### What version of SQLBoiler are you using (`sqlboiler --version`)?
v4.8.6

For the latest language standards, it should not be used: if err == nil {}

```
	if len(cache.retMapping) != 0 {
		err = exec.QueryRowContext(ctx, cache.query, vals...).Scan(returns...)
		if err == sql.ErrNoRows {
			err = nil // Postgres doesn't return anything when there's no update
		}
	} else {
		_, err = exec.ExecContext(ctx, cache.query, vals...)
	}
```

**Comment by stephenafamo** [comment]
That makes sense.

If you can send a PR for this it would be greatly appreciated.

**Comment by stephenafamo** [comment]
Fixed in `v4.9.0`
